recipes/mfem/meta.yaml

```diff
@@ -0,0 +1,53 @@
+{% set name = "mfem" %}
+{% set version = "4.2" %}
+
+{% set mpi = mpi or 'mpich' %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/mfem/mfem/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 15ff770bb97565b085f0b8b91df60b529f427c134d1d3fe7f700aca9d3209884
+
+build:
+  number: 0
+  skip: true  # [not linux]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - make
+    - {{ mpi }}  # [build_platform != target_platform and mpi == "openmpi"]
+  host:
+    - {{ mpi }}
+    - hypre
+    - metis
+  run:
+    - {{ mpi }}
+    - hypre
+    - metis
+
+test:
+  commands:
+    - test -f $PREFIX/lib/libmfem.a
```

Comment:
Can you build shared libraries instead of static ones?

Reply:
We'd like to avoid static libraries. Are they needed?

Reply:
Only if someone wants to build statically linked executables against libmfem. I'm not certain how common that is.

Reply:
Let's remove them for now and add them if they are needed. static libraries are discouraged, because the downstream packages don't get updates, downstream packages need to repackage the upstream package's license and they increase the size of the package.

Reply:
I removed the static build.